--- pygtk-shutdown	2011-06-12 20:09:18.000000000 -0700
+++ pygtk-shutdown-hal	2011-06-12 20:09:56.000000000 -0700
@@ -102,10 +102,10 @@
         self.callcmd(LOGOUT_COMMAND)
 
     def shutdown(self, widget = None, data = None):
-        self.callcmd("/sbin/shutdown -h now")
+        self.callcmd("dbus-send --system --dest=org.freedesktop.Hal --print-reply /org/freedesktop/Hal/devices/computer org.freedesktop.Hal.Device.SystemPowerManagement.Shutdown")
 
     def reboot(self, widget = None, data = None):
-        self.callcmd("/sbin/shutdown -r now")
+        self.callcmd("dbus-send --system --dest=org.freedesktop.Hal --print-reply /org/freedesktop/Hal/devices/computer org.freedesktop.Hal.Device.SystemPowerManagement.Reboot")
 
     def callcmd(self, cmd):
         proc = subprocess.Popen(cmd, stdout = subprocess.PIPE, stderr =
@@ -123,7 +123,7 @@
                 subprocess.Popen("kill -9 " + str(proc.pid), shell = True)
 
         if status != 0:
-            subprocess.Popen("gksudo -- " + cmd, shell = True)
+            subprocess.Popen(cmd, shell = True)
 
         self.terminate()
 
